gcsio/src/test/java/com/google/cloud/hadoop/gcsio/InvocationIdGrpcChannelInterceptorTest.java

```diff
@@ -0,0 +1,131 @@
+package com.google.cloud.hadoop.gcsio;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.assertSame;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import io.grpc.CallOptions;
+import io.grpc.Channel;
+import io.grpc.ClientCall;
+import io.grpc.ClientInterceptors;
+import io.grpc.Metadata;
+import io.grpc.MethodDescriptor;
+import io.grpc.testing.TestMethodDescriptors;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.ArgumentMatchers;
+
+@RunWith(JUnit4.class)
+public class InvocationIdGrpcChannelInterceptorTest {
+
+  private InvocationIdGrpcChannelInterceptor invocationIdInterceptor;
+
+  private Channel channel = mock(Channel.class);
```

Comment:
is there a way to avoid mock? We try to avoid mock as much as possible.

Reply:
Not sure why? In UT's we should mock all the dependent calls and just test the `unit` isn't it?

Reply:
Mocks are hard to maintain. Refer GoogleCloudStorageGrpcReadChannelTest. We are using FakeService. Why cant we do something similar?